### ISSUE TITLE
Don't refresh ttl on cached keys

### ISSUE THREAD

**Issue by chrisskilton** [sindresorhus/got#603]
I've got a basic`got` request setup with a redis keyv instance as the cache:

```
got.stream(target, {cache});
```

If i make a request to `/somepath` the data gets cached in redis with the ttl I set on the keyv instance, which is great.

If I make a _second_ request to `/somepath` within the key's ttl the data gets returned from the cache, which is also great. 

The problem is the second request _reset_ the ttl in the cache. So if requests kept coming in the data will never be cleared from the cache, which is bad (for my use case).

Is there an existing bit of config that I'm missing for this behaviour? I've tried setting the `max-age` header and various other `cache-control` headers but feel I'm missing something!

Any help would be greatly appreciated

**Comment by szmarczak** [comment]
`cacheable-request` nor `keyv` resets TTL when retrieving an element. Redis does not too.

Could you please include a full example?

**Comment by chrisskilton** [comment]
@szmarczak sure thing. thanks for the response. here's some example code:

test server - this will increment an integer and return the string on every call to it
```javascript
const http = require('http');

let response = 0;

const server = http.createServer((req, res) => {
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end(response.toString());
    response++;
});

server.listen(3000);
```

Test - this will use a regular `keyv` instance as a cache with `got`, using a ttl of 3000
```javascript
const got = require('got');
const Keyv = require('keyv');
const cache = new Keyv({
    ttl: 3000
});

const fetch = async () => {
    const response = await got('http://localhost:3000', {cache});

    console.log('Served from cache:', response.fromCache);
    console.log('Response:', response.body);
};

const wait = () => {
    console.log('waiting 1s...');
    return new Promise(res => setTimeout(res, 1000));
};

const run = async () => {
    //first call should be a cache miss
    await fetch();
    await wait();

    //1s after first call should be a cache hit
    await fetch();
    await wait();

    //2s after first call should be a cache hit
    await fetch();
    await wait();

    //3s after first call should be a cache miss again
    await fetch();
    await wait();

    //4s should be a cache hit
    await fetch();
    await wait();

    //5s should be a cache hit
    await fetch();
    await wait();
};

run();
```

Each call after the initial cacheing resets the cached value ttl to 3s again. 

Console output:
```
Served from cache: false
Response: 0
waiting 1s...
Served from cache: true
Response: 0
waiting 1s...
Served from cache: true
Response: 0
waiting 1s...
Served from cache: true
Response: 0
waiting 1s...
Served from cache: true
Response: 0
waiting 1s...
Served from cache: true
Response: 0
waiting 1s...
```

As you can see after the first call, if there isn't a full 3s wait, the cached data isn't ever cleared.

**Comment by chrisskilton** [comment]
got a feeling I'm making the same assumption you did here https://github.com/sindresorhus/got/pull/284#issuecomment-300193765

**Comment by szmarczak** [comment]
Could you try Got 9.2.1?

**Comment by szmarczak** [comment]
Could you try setting cache control (max age) to 3 seconds and try with 9.2.1?

**Comment by szmarczak** [comment]
> the ttl is purely for the key/value store to be able to clean up after itself, not any sort of control over how long you want to cache any particular responses.

No, it just sets default TTL if server didn't specify.

There was a bug in Got and it's fixed in 9.2.1.
